core/Plugin/Controller.php

```diff
@@ -476,25 +476,18 @@ protected function getLastUnitGraphAcrossPlugins($currentModuleName, $currentCon
      */
     protected function getGraphParamsModified($paramsToSet = array())
     {
-        if (!isset($paramsToSet['period'])) {
-            $period = Common::getRequestVar('period');
-        } else {
-            $period = $paramsToSet['period'];
+        try {
+            $period = isset($paramsToSet['period']) ?? Piwik::getPeriod();
+        } catch (\Exception $e) {
+            $period = null;
         }
```

Comment:
I don't think this try catch is useful here. If no period was provided, the graph can't be displayed. So throwing an exception, like it was before, should be fine.